framework/include/updater.hpp

```diff
@@ -4,9 +4,26 @@
 
 namespace gvk
 {
-	class updater;
+	class updater;	
 	using event_t = std::variant<std::shared_ptr<event>, files_changed_event, swapchain_changed_event, swapchain_resized_event>;
-	using updatee_t = std::variant<avk::graphics_pipeline, avk::compute_pipeline, avk::ray_tracing_pipeline, avk::image, avk::image_view>;
+	using listener_updatee_t = std::function<void(void)>;
+	using resource_updatee_t = std::variant<avk::graphics_pipeline, avk::compute_pipeline, avk::ray_tracing_pipeline, avk::image, avk::image_view>;
+	using updatee_t = std::variant<avk::graphics_pipeline, avk::compute_pipeline, avk::ray_tracing_pipeline, avk::image, avk::image_view, listener_updatee_t>;
+
+
+	/**
+	* concept used to allow variadic templates used only on resource_updatee_t type
+	*/
+	template <typename T>
+	concept is_updatee = requires (T& aT, resource_updatee_t& aVar) {
+		aVar = aT;
+	};
+
+	/**
+	* concept used to allow variadic templates used only on is_callee type
+	*/
+	template <typename T>
+	concept is_callee = std::same_as<T, listener_updatee_t>;
```

Comment:
The biggest problem I have with these callees/listener_updatees is that they require the user to pass-in `std::function<void(void)>`. This is a very unnecessary restriction. Storing them INTERNALLY as `std::function<void(void)>` is fine, but the user should not be restricted to passing-in that type! 

Maybe see if you could put `std::convertible_to` to good use here instead of `std::same_as`!

Reply:
nice!